cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -2673,6 +2610,72 @@ mod test {
         assert!(reg.entries.is_empty());
     }
 
+    #[test]
+    fn prunes_branches_on_delete() {
+        let (flox, _temp_dir_handle) = flox_instance();
+
+        let env1_dir = flox.temp_dir.join("env1");
+        std::fs::create_dir_all(&env1_dir).unwrap();
+        let env1_dir = CanonicalPath::new(env1_dir).unwrap();
+
+        let env2_dir = flox.temp_dir.join("env2");
+        std::fs::create_dir_all(&env2_dir).unwrap();
+        let env2_dir = CanonicalPath::new(env2_dir).unwrap();
+
+        // dummy paths since we are not rendering the environment
+        let rendered_env_links =
+            RenderedEnvironmentLinks::new_unchecked(PathBuf::new(), PathBuf::new());
+
+        // create a mock remote
+        let (test_pointer, remote_path, remote) = create_mock_remote(flox.temp_dir.join("remote"));
+
+        let remote_branch = remote_branch_name(&test_pointer);
+        remote.checkout(&remote_branch, true).unwrap();
+        commit_file(&remote, "file 1");
+
+        let env1_branch = branch_name(&test_pointer, &env1_dir);
+        let env2_branch = branch_name(&test_pointer, &env2_dir);
+
+        // Create a mock floxmeta
+        let floxmeta = create_floxmeta(&flox, &remote_path, &test_pointer, &remote_branch);
+
+        // Create two environments that use the same pointer.
+        let env1 = ManagedEnvironment::open_with(
+            floxmeta.clone(),
+            &flox,
+            test_pointer.clone(),
+            env1_dir,
+            rendered_env_links.clone(),
+        )
+        .unwrap();
+        let env2 = ManagedEnvironment::open_with(
+            floxmeta.clone(),
+            &flox,
+            test_pointer.clone(),
+            env2_dir,
+            rendered_env_links.clone(),
+        )
+        .unwrap();
+
+        assert!(floxmeta.git.has_branch(&remote_branch).unwrap());
+        assert!(floxmeta.git.has_branch(&env1_branch).unwrap());
+        assert!(floxmeta.git.has_branch(&env2_branch).unwrap());
+
+        env2.delete(&flox).unwrap();
+
+        // Only env2 should be pruned.
+        assert!(floxmeta.git.has_branch(&remote_branch).unwrap());
+        assert!(floxmeta.git.has_branch(&env1_branch).unwrap());
+        assert!(!floxmeta.git.has_branch(&env2_branch).unwrap());
+
+        env1.delete(&flox).unwrap();
```

Comment:
suggestion followup: can we call gc or prune_nonexistent here? That seems more in line with your comment demonstrating testing this with real use